src/hypothesis/internal/conjecture/engine.py

```diff
@@ -504,6 +572,13 @@ def _run(self):
                         i += k
                 k //= 2
 
+            self.zero_blocks()
+
+            minimize(
```

Comment:
That is a good question to which I no longer remember the exact reasoning.

I think the idea was simply that it's a useful thing to do early on and it's now much cheaper so it's safe to do it here.